app/src/main/java/org/simple/clinic/teleconsultlog/medicinefrequency/MedicineFrequencyBottomSheet.kt

```diff
@@ -0,0 +1,131 @@
+package org.simple.clinic.teleconsultlog.medicinefrequency
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.sheet_medicine_frequency.*
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.R
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.teleconsultlog.drugduration.DrugDuration
+import org.simple.clinic.teleconsultlog.medicinefrequency.di.MedicineFrequencyComponent
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.BottomSheetActivity
+import javax.inject.Inject
+
+class MedicineFrequencyBottomSheet : BottomSheetActivity(), MedicineFrequencySheetUiActions {
+
+  @Inject
+  lateinit var effectHandlerFactory: MedicineFrequencyEffectHandler.Factory
+
+  private lateinit var component: MedicineFrequencyComponent
+
+  companion object {
+    private const val DRUG_DURATION = "drugDuration"
+    private const val MEDICINE_FREQUENCY = "medicineFrequency"
+    private const val SAVED_MEDICINE_FREQUENCY = "savedmedicineFrequency"
+
+    fun intent(
+        context: Context,
+        drugDuration: DrugDuration,
+        medicineFrequency: MedicineFrequencies
+    ): Intent {
+      return Intent(context, MedicineFrequencyBottomSheet::class.java).apply {
+        putExtra(DRUG_DURATION, drugDuration)
+        putExtra(MEDICINE_FREQUENCY, medicineFrequency)
+      }
+    }
+  }
+
+  private val drugDuration by unsafeLazy {
+    intent.getParcelableExtra<DrugDuration>(DRUG_DURATION)!!
+  }
+
+  private val medicineFrequency by unsafeLazy {
+    intent.getSerializableExtra(MEDICINE_FREQUENCY)
+  }
+
+  private val events by unsafeLazy {
+    saveClicks()
+  }
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forActivity(
+        events = events.ofType(),
+        defaultModel = MedicineFrequencyModel.create(MedicineFrequencies.BD),
```

Comment:
It seems like we are not tracking radio button check changes anywhere, so if we are doing state restoration the screen will go back to the default `MedicineFrequency` that is passed to the screen instead of remembering user selection. Please, take a look at it.